plugins/source_testing.go

```diff
@@ -75,9 +76,36 @@ func validateTables(t *testing.T, tables schema.Tables, resources []*schema.Reso
 
 func validateResource(t *testing.T, resource *schema.Resource) {
 	t.Helper()
+	// we want to marshal and unmarshal to mimic over-the-wire behavior
+	b, err := json.Marshal(resource.Data)
+	if err != nil {
+		t.Fatalf("failed to marshal resource data: %v", err)
+	}
+	var data map[string]interface{}
+	if err := json.Unmarshal(b, &data); err != nil {
+		t.Fatalf("failed to unmarshal resource data: %v", err)
+	}
 	for _, columnName := range resource.Table.Columns.Names() {
-		if resource.Get(columnName) == nil && !resource.Table.Columns.Get(columnName).IgnoreInTests {
+		if data[columnName] == nil && !resource.Table.Columns.Get(columnName).IgnoreInTests {
 			t.Errorf("table: %s with unset column %s", resource.Table.Name, columnName)
 		}
+		val := data[columnName]
+		if val != nil {
+			switch resource.Table.Columns.Get(columnName).Type {
+			case schema.TypeJSON:
```

Comment:
I think there's also going to be a new class of bug where we set a JSON column using a string or byte slice (that was already marshalled to JSON). It's very rare, if ever, that we'd want to write a string directly to a JSON column, even though it would marhsal to JSON just fine, so maybe we can check for that issue here